### PR TITLE
fix quickstart dns sans 

### PR DESCRIPTION
**Pull request by qrkourier**

### CODE REVIEW

File: .github/workflows/test-quickstart.yml [hunk 1]
```diff
@@ -1,16 +1,12 @@
...
   push:
     branches:
       - main
-    paths:
-      - 'quickstart/**'
-  # test quickstart changes before merge
   pull_request:
-    paths:
-      - 'quickstart/**'
+    branches:
```
Comment (qrkourier) [inline .github/workflows/test-quickstart.yml:8]: stop waiting for quickstart files to change so this runs earlier in the ziti life cycle

File: quickstart/docker/image/run-router.sh [hunk 2]
```diff
@@ -1,5 +1,9 @@
 #!/bin/bash
 
+set -o errexit
```
Comment (qrkourier) [inline quickstart/docker/image/run-router.sh:3]: raise uncaught exceptions

File: quickstart/docker/image/run-router.sh [hunk 3]
```diff
@@ -32,6 +36,9 @@ if [[ "${_ZITI_ROUTER_NAME}" != "" ]]; then
 fi
 
 _CONFIG_PATH="${ZITI_HOME}/${ZITI_ROUTER_NAME}.yaml"
+# Set an error trap to move _CONFIG_PATH when it's unsafe to assume enrollment succeeded
+trap '[[ -f "${_CONFIG_PATH}" ]] && mv "${_CONFIG_PATH}" "${_CONFIG_PATH}.err"' ERR
```
Comment (qrkourier) [inline quickstart/docker/image/run-router.sh:40]: make the script more resilient against bad state problems resulting from an incomplete run

File: quickstart/docker/image/run-router.sh [hunk 4]
```diff
@@ -55,8 +62,8 @@ if [ ! -f "${_CONFIG_PATH}" ]; then
     createPrivateRouterConfig "${ZITI_ROUTER_NAME}"
   fi
 
-  found=$("${ZITI_BIN_DIR-}/ziti" edge list edge-routers 'name = "'"${ZITI_ROUTER_NAME}"'"' | grep -c "${ZITI_ROUTER_NAME}")
-  if [[ found -gt 0 ]]; then
```
Comment (qrkourier) [inline quickstart/docker/image/run-router.sh:59]: `[[ found -gt 0 ]]` is always false

File: quickstart/docker/image/ziti-cli-functions.sh [hunk 5]
```diff
@@ -784,17 +784,18 @@ function createPki {
   _pki_create_intermediate "${ZITI_GRANDPARENT_INTERMEDIATE}" "${ZITI_PKI_SIGNER_INTERMEDIATE_NAME}" 1
 
   echo " "
-  pki_allow_list="localhost,${ZITI_NETWORK}"
+  ctrl_pki_allow_list="localhost,${ZITI_NETWORK}"
```
Comment (qrkourier) [inline quickstart/docker/image/ziti-cli-functions.sh:787]: keep ctrl and edge DNS SANs separate

**Review by dovholuknf — APPROVED** [review]

**Pull request merged**